### PR TITLE
Proj/bonza initial

### PR DESCRIPTION
**Pull request by mogii**

### CODE REVIEW

**Comment by mogii** [comment]
@shintre Need to ignore node_modules

**Comment by shintre** [comment]
@mogii node_modules removed and added .gitignore

**Comment by shintre** [comment]
@mogii 
+ I think this is now ready to merge in the main branch
+ Added test cases.

File: .gitignore
Comment (mogii) [inline .gitignore:1]: .github/ shouldn't be ignored, it contains the workflow settings. @shintre

Comment (shintre) [reply .gitignore:1]: done

**Pull request merged**